iampolicymanagementv1/iam_policy_management_v1_examples_test.go

```diff
@@ -676,5 +681,400 @@ var _ = Describe(`IamPolicyManagementV1 Examples Tests`, func() {
 			Expect(response.StatusCode).To(Equal(204))
 
 		})
+		It(`ListPolicyTemplates request example`, func() {
+			fmt.Println("\nListPolicyTemplates() result:")
+			// begin-list_policy_templates
+
+			listPolicyTemplatesOptions := iamPolicyManagementService.NewListPolicyTemplatesOptions(
+				exampleAccountID,
+			)
+
+			policyTemplateCollection, response, err := iamPolicyManagementService.ListPolicyTemplates(listPolicyTemplatesOptions)
+			if err != nil {
+				panic(err)
+			}
+			b, _ := json.MarshalIndent(policyTemplateCollection, "", "  ")
+			fmt.Println(string(b))
+
+			// end-list_policy_templates
+
+			Expect(err).To(BeNil())
+			Expect(response.StatusCode).To(Equal(200))
+			Expect(policyTemplateCollection).ToNot(BeNil())
+		})
+		It(`CreatePolicyTemplate request example`, func() {
+			fmt.Println("\nCreatePolicyTemplate() result:")
+			// begin-create_policy_template
+
+			policyRole := &iampolicymanagementv1.Roles{
+				RoleID: core.StringPtr("crn:v1:bluemix:public:iam::::role:Viewer"),
+			}
+			v2PolicyGrant := &iampolicymanagementv1.Grant{
+				Roles: []iampolicymanagementv1.Roles{*policyRole},
+			}
+			v2PolicyControl := &iampolicymanagementv1.Control{
+				Grant: v2PolicyGrant,
+			}
+			serviceNameResourceAttribute := &iampolicymanagementv1.V2PolicyResourceAttribute{
+				Key:      core.StringPtr("serviceName"),
+				Operator: core.StringPtr("stringEquals"),
+				Value:    core.StringPtr("iam-access-management"),
+			}
+			policyResource := &iampolicymanagementv1.V2PolicyResource{
+				Attributes: []iampolicymanagementv1.V2PolicyResourceAttribute{
+					*serviceNameResourceAttribute},
+			}
+
+			templatePolicyModel := &iampolicymanagementv1.TemplatePolicy{
+				Type:        core.StringPtr("access"),
+				Description: core.StringPtr("Test Template"),
+				Resource:    policyResource,
+				Control:     v2PolicyControl,
+			}
+
+			createPolicyTemplateOptions := iamPolicyManagementService.NewCreatePolicyTemplateOptions(
+				examplePolicyTemplateName,
+				exampleAccountID,
+				templatePolicyModel,
+			)
+
+			policyTemplate, response, err := iamPolicyManagementService.CreatePolicyTemplate(createPolicyTemplateOptions)
+			if err != nil {
+				panic(err)
+			}
+			b, _ := json.MarshalIndent(policyTemplate, "", "  ")
+			fmt.Println(string(b))
+
+			// end-create_policy_template
+
+			Expect(err).To(BeNil())
+			Expect(response.StatusCode).To(Equal(201))
+			Expect(policyTemplate).ToNot(BeNil())
+
+			examplePolicyTemplateID = *policyTemplate.ID
```

Comment:
If you define the appropriate link objects in your API definition, the SDK generator will generate code like this to store off a value such as the policyTemplate.ID value, then generate code in the test of the consuming operation to use the stored value to initialize an operation parameter.
Take a look at the SDK squad's mock service that we use for various types of testing:
https://github.ibm.com/CloudEngineering/openapi-sdkgen/blob/main/src/test/resources/ansiform-service.yaml
and the associated Go code in the resulting Go SDK:
https://github.ibm.com/CloudEngineering/ansiform-go-sdk/tree/main/ansiformmockv1

Note that the integration tests and examples code were generated for this service with no required manual post-gen changes.  This is due to the fact that example values and links are defined throughout the API definition.